src/storage/v2/storage.hpp

```diff
@@ -394,6 +399,18 @@ class Storage {
   Indices indices_;
   Constraints constraints_;
 
+  // Datastructures to provide fast retrieval of node-label and
+  // edge-type related metadata.
+  // Currently we should not remove any node-labels or edge-types even
+  // if the set of given types are currently not present in the
```

Comment:
Why do we need to keep labels/types which are not present now in graph?

Reply:
We do not reference count their usage, hence no way to deterministically remove their validity upon removal of last usage.

Reply:
Just as @Ignition  said.
Providing a solution that keeps track of the count of each element is currently not possible, and I would rather not add an atomic RMW op to every single call that creates a label/edgetype.